roxie/ccd/ccdqueue.cpp

```diff
@@ -2825,7 +2842,19 @@ class RoxieSocketQueueManager : public RoxieReceiverBase
                         {
                             StringBuffer xx; logctx.CTXLOG("Retry %d received on subchannel %u for %s", retries+1, mySubchannel, header.toString(xx).str());
                         }
-                        queue.enqueueUnique(packet.getClear(), mySubchannel, 0);  // MORE - what about IBYTI processing in NEW_IBYTI mode
+#ifdef NEW_IBYTI
+                        // It's debatable whether we should delay for the primary here - they had one chance already...
+                        // But then again, so did we, assuming the timeout is longer than the IBYTIdelay
+                        if (mySubchannel != 0)  // i.e. I am not the primary here
+                        {
+                            unsigned delay = 0;
+                            for (unsigned subChannel = 0; subChannel < mySubchannel; subChannel++)
+                                delay += getIbytiDelay(header.subChannels[subChannel]);
+                            delayed.queryQueue(header.channel, mySubchannel).append(packet.getClear(), msTick()+delay);
```

Comment:
Not a new change - but if delay==0 this should probably add directly to the real queue (and in the other location).  Could easily restructure to conditionally calc delay and then conditional on delay